crates/wasmtime/src/linker.rs

```diff
@@ -270,6 +271,117 @@ impl Linker {
         Ok(self)
     }
 
+    /// Define automatic instantiations of a [`Module`] in this linker.
+    ///
+    /// For a Reactor module, this instantiates the module and adds all its
+    /// exports to the `Linker` as the `instance` method.
+    ///
+    /// For a Command module, this adds exports to the `Linker` which
+    /// instantiate the module each time the exports are called.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the any item is redefined twice in this linker (for
+    /// example the same `module_name` was already defined) and shadowing is
+    /// disallowed, or if `instance` comes from a different [`Store`] than this
+    /// [`Linker`] originally was created with.
```

Comment:
I think this section may want to also get expanded with information about how the `Linker` needs to already contain all items necessary to instantiate the module when this function is called, even if this is a command.